llvm/utils/TableGen/GlobalISelCombinerEmitter.cpp

```diff
@@ -2459,6 +2459,7 @@ void GICombinerEmitter::emitRunCustomAction(raw_ostream &OS) {
     OS << "  switch(ApplyID) {\n";
     for (const auto &Apply : ApplyCode) {
       OS << "  case " << Apply->getEnumNameWithPrefix(CXXApplyPrefix) << ":{\n"
+         << "    Helper.getBuilder().setInstrAndDebugLoc(*State.MIs[0]);\n"
```

Comment:
I suppose `State.MIs[0]` is always the current instruction that is being tried to be combined. I ran all tests and it works good so far.